src/StructuredLogger/BinaryLogger/Postprocessing/StringReadEventArgs.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Build.Logging
+{
+    /// <summary>
+    /// An event args for <see cref="IBuildEventStringsReader.StringReadDone"/> callback.
+    /// </summary>
+    public sealed class StringReadEventArgs : EventArgs

```

Comment:
One of my more serious concerns is that this is a class, and we'll allocate it for every string (but only if someone is listening). The binlog I'm currently looking at has 2.2 million strings, at 40 bytes per object on 64-bit it'll be around 100 MB allocations right there.

Wondering if we can find a way to avoid allocating per string here, but if it's hard, it's OK. As long as we make sure the normal path of opening binlogs is not impacted negatively.

I've worked extremely hard to make sure binlog opening is as fast as possible.

Reply:
Right. I definitely have an appreciation for any perf efforts :-)

In this case there is a single instance of the type per entire log reading and it's being reused between event calls: https://github.com/KirillOsenkov/MSBuildStructuredLog/pull/711/files#diff-5850999507dea96dddb234354cf875f2e8ba46b9f7d0e0dd1019bbf137e3e60bR1558

The situation is explicitly documented on the event declaration:
https://github.com/KirillOsenkov/MSBuildStructuredLog/pull/711/files#diff-6f29c5341d4c9e3afce3ea5138039ac5d6aee77639c0744de193a7a7b9c6ed5dR16

```
/// The passed event arg can be reused and should not be stored.
```